packages/material-ui/src/Radio/Radio.js

```diff
@@ -91,7 +91,12 @@ const Radio = React.forwardRef(function Radio(inProps, ref) {
 
   if (radioGroup) {
     if (typeof checked === 'undefined') {
-      checked = radioGroup.value === props.value;
+      if (typeof radioGroup.value === 'object' && typeof props.value === 'object') {
+        checked = radioGroup.value === props.value;
```

Comment:
There is one test that check the value as object, I am not sure why it is added so I am keeping it to not break the existing behaviour.

Reply:
No objections to removing `typeof object` (and updating the test) as it breaks the narrative I have been pushing for saying that value should be of type of any, as any structure with `.toString()` should work.